tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py

```
@@ -72,14 +76,62 @@ def oauth_user_info(self, provider, response=None):
                 "role_keys": user_roles,
             }
 
-    def get_oauth_token(self, token=None):
+    def get_oauth_token(self):
         """
         Retrieves the oauth token from the session.
+        If the access token is expired, it will try to refresh it using the refresh token.
+        Returns an empty dict if there is no session or the token couldn't be refreshed.
+        """
+        token = session.get("oauth_token", {})
+
+        if token:
+            # Check if the token is expired
+            if self.is_token_expired(token):
+                # Try to refresh the token
+                try:
+                    new_token = self.refresh_token(token)
+                except OAuthError as e:
+                    logging.error("Failed to refresh the token: {}".format(e))
+                    new_token = {}
```

It would be better to throw the error and remove the log